Update README (#373)
JuliaGaussianProcesses/KernelFunctions.jl: README.md

## Kernel functions for machine learning

> KernelFunctions.jl provide a flexible and complete framework for kernel functions.

Reviewer: ```suggestion
KernelFunctions.jl provides a flexible and complete framework for kernel functions.
```
nb- "complete" in what sense ?

Author: A good question. I suspect @theogf wrote this originally, so probably the person to ask.

Reviewer 2: 🤷

Reviewer 2: I think I mean `extensive` or some word meaning that we have a lot of features!

Author: How about something like
```suggestion
KernelFunctions.jl provides a flexible framework for defining kernel functions, and an extensive collection of implementations.
```

> The aim is to make the API as model-agnostic as possible while still being user-friendly.

Reviewer: While we're looking at it: should we mention here that KernelFunctions.jl does not handle parameters by itself (ensuring constraints such as positivity, or getting its gradients) and point to other packages such as ParameterHandling.jl and Flux's parameters for how to do that?

Reviewer 3: I was planning to add an example on parameter handling to the docs over the next few days. I have been experimenting, and should be easy to clean it up and make a PR.

Author: > I was planning to add an example on parameter handling to the docs over the next few days. I have been experimenting, and should be easy to clean it up and make a PR.

Nice. I look forward to it

> While we're looking at it: should we mention here that KernelFunctions.jl does not handle parameters by itself (ensuring constraints such as positivity, or getting its gradients) and point to other packages such as ParameterHandling.jl and Flux's parameters for how to do that?

Good point. Would something like
```suggestion
The aim is to make the API as model-agnostic as possible while still being user-friendly, and to interoperate well with generic packages for handling parameters like ParameterHandling.jl and Functors.jl.
```
be what you're after @st-- ?

Reviewer 2: Looks great! Maybe you could also include the links to the packages

Author: ```suggestion
The aim is to make the API as model-agnostic as possible while still being user-friendly, and to interoperate well with generic packages for handling parameters like [ParameterHandling.jl](https://github.com/invenia/ParameterHandling.jl/) and FluxML's [Functors.jl](https://github.com/FluxML/Functors.jl/).
```

  <img src="docs/src/assets/heatmap_combination.png" width=400px>
</p>

## Related Work

Directly inspired by the [MLKernels](https://github.com/trthatcher/MLKernels.jl) package.

> See the JuliaGaussianProcesses [github organisation](https://github.com/JuliaGaussianProcesses) and [website](https://juliagaussianprocesses.github.io/) for more related packages.

Reviewer 2: ```suggestion
See the JuliaGaussianProcesses [Github organisation](https://github.com/JuliaGaussianProcesses) and [website](https://juliagaussianprocesses.github.io/) for more related packages.
```
